mypy/types.py

```diff
@@ -360,6 +360,9 @@ def __repr__(self) -> str:
         else:
             return "NotRequired[{}]".format(self.item)
 
+    def accept(self, visitor: 'TypeVisitor[T]') -> T:
```

Comment:
This feels icky, but I presume it's needed? 

I think I would have preferred fixing this one layer up -- whatever is trying to figure out if this is a Required should maybe know (this only works if that one layer up is related to TypedDicts which it probably isn't).

Reply:
The caller is this function, https://github.com/python/mypy/blob/master/mypy/typeanal.py#L1304. It doesn't feel like that function should need to have knowledge of typeddicts/required.

The caller above that  is https://github.com/python/mypy/blob/master/mypy/semanal_typeddict.py#L305. This does look like a reasonable place. If you would prefer I have the typeddict semantic analyzer strip Required/NotRequired before calling explicit any check I can do that.

Reply:
I think either approach works:
 - Your current approach is more broad and will generally fix the problem with accepting a Required anywhere, as long as just having the item accept it is what is always meant (it sounds like it might be).
 - Just changing the function above is less broad but also has less chance of accidentally introducing subtly wrong behavior.

I think the "subtly wrong behavior" is so unlikely that whatever approach is easier will work.